source/RenderBuffer.cpp

```diff
@@ -0,0 +1,229 @@
+	glGenTextures(1, &m_texid);
+	glBindTexture(GL_TEXTURE_2D, m_texid);
+
+	// Use linear interpolation and no wrapping.
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+
+	// Attach a blank image to the texture
+	glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, m_size.X(), m_size.Y(), 0, GL_RGBA, GL_UNSIGNED_BYTE, nullptr);
+
+	// Attach the texture to the frame buffer
+	glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, m_texid, 0);
+	GLenum draw_buffers[] = {GL_COLOR_ATTACHMENT0};
+	glDrawBuffers(1, draw_buffers);
+
+	if(glCheckFramebufferStatus(GL_FRAMEBUFFER) != GL_FRAMEBUFFER_COMPLETE)
+	{
+		Logger::LogError("Failed to initialize framebuffer for RenderBuffer");
+	}
+
+
+	glBindTexture(GL_TEXTURE_2D, 0);
+	glBindFramebuffer(GL_FRAMEBUFFER, 0);
+
+	// default to the current viewport size at the time of construction
+	glGetIntegerv(GL_VIEWPORT, m_last_viewport);
+
+	// save off the screen size for the game coordinates (not the same as
+	// the window size or the viewport size)
+	m_old_width = Screen::Width();
+	m_old_height = Screen::Height();
+}
+
+
+
+// Destructor. Frees the texture and renderbuffers
+RenderBuffer::~RenderBuffer()
+{
+	glDeleteTextures(1, &m_texid);
+	glDeleteFramebuffers(1, &m_framebuffer);
+}
+
+
+
+// Turn this buffer on as a render target. The render target is restored if
+// the Activation object goes out of scope.
+RenderBuffer::Activation RenderBuffer::SetTarget()
+{
+	// NOTE: These glGets can cause an unwanted state synchronization that can
+	//       cause performance problems. The only real reason we might want this
+	//       is if we are nesting render buffers. If only one framebuffer is
+	//       enabled at a time, then we can just reset the buffer to 0 when we
+	//       are done.
+	glGetIntegerv(GL_FRAMEBUFFER_BINDING, reinterpret_cast<int*>(&m_last_framebuffer));
+	glGetIntegerv(GL_VIEWPORT, m_last_viewport);
+	glBindFramebuffer(GL_FRAMEBUFFER, m_framebuffer);
+
+
+	glViewport(0, 0, m_size.X(), m_size.Y());
+	m_old_width = Screen::Width();
+	m_old_height = Screen::Height();
+
+	Screen::SetDimensionsInternal(m_size.X(), m_size.Y());
+
+	static const float CLEAR[] = {0, 0, 0, 0};
+	glClearBufferfv(GL_COLOR, 0, CLEAR);
+
+	return Activation(*this);
+}
+
+
+
+// Reset the render target and viewport to the original settings
+void RenderBuffer::Deactivate()
+{
+	// Restore the old settings
+	Screen::SetDimensionsInternal(m_old_width, m_old_height);
+	glViewport(m_last_viewport[0], m_last_viewport[1], m_last_viewport[2], m_last_viewport[3]);
+	glBindFramebuffer(GL_FRAMEBUFFER, m_last_framebuffer);
+}
+
+
+
+// Draw the contents of this buffer at the specified position.
+void RenderBuffer::Draw(const Point& position, const Point& clipsize, const Point& srcposition)
```

Comment:
```suggestion
void RenderBuffer::Draw(const Point &position, const Point &clipsize, const Point &srcposition)
```
Why is your formatting so inconsistent?

Reply:
Every other project I have worked on requires me to put the reference with the type, not the variable. Its a hard habit to break. 

The ones that I have fixed are due to the `check_code_style.py` script flagging them, but it didn't catch this one for some reason.

Reply:
> Every other project I have worked on requires me to put the reference with the type, not the variable. Its a hard habit to break.
> 
> The ones that I have fixed are due to the `check_code_style.py` script flagging them, but it didn't catch this one for some reason.

No, there are a ton more instances of this. The last I checked I had 20 of these reviewed but I decided not to include them in the review for terseness.

Reply:
Hopefully I found them all.